Shared/Samples/Play KML tour/PlayKMLTourView.swift

```diff
@@ -0,0 +1,143 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct PlayKMLTourView: View {
+    /// A scene with an imagery basemap and a world elevation service.
+    @State private var scene: ArcGIS.Scene = {
+        let scene = Scene(basemapStyle: .arcGISImagery)
+        scene.initialViewpoint = Viewpoint(latitude: .nan, longitude: .nan, scale: 114_145_911)
+        
+        // Set the scene's base surface with an elevation source.
+        let surface = Surface()
+        let elevationSource = ArcGISTiledElevationSource(url: .worldElevationService)
+        surface.addElevationSource(elevationSource)
+        scene.baseSurface = surface
+        
+        return scene
+    }()
+    
+    /// The current viewpoint of the scene view used when reseting the tour.
+    @State private var viewpoint: Viewpoint?
+    
+    /// The tour controller used to control the KML tour.
+    @State private var tourController = KMLTourController()
+    
+    /// The current status of the KML tour.
+    @State private var tourStatus: KMLTour.Status = .notInitialized
+    
+    /// A Boolean that indicates whether to show an error alert.
+    @State private var isShowingErrorAlert = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error? {
+        didSet { isShowingErrorAlert = error != nil }
+    }
+    
+    /// A Boolean value that indicates whether to disable the tour buttons.
+    private var tourDisabled: Bool {
+        tourStatus == .notInitialized || tourStatus == .initializing
+    }
+    
+    var body: some View {
+        // Create a scene view with a scene and a viewpoint.
+        SceneView(scene: scene, viewpoint: viewpoint)
+            .onViewpointChanged(kind: .centerAndScale) { viewpoint = $0 }
+            .task {
+                do {
+                    // Add KML layer to the scene.
+                    let dataset = KMLDataset(url: .esriTour)
+                    try await dataset.load()
+                    let layer = KMLLayer(dataset: dataset)
+                    scene.addOperationalLayer(layer)
+                    
+                    // Set the tour controller with first tour in the dataset.
+                    tourController.tour = dataset.tours.first
+                    
+                    // Listen for tour status updates.
+                    if let tour = tourController.tour {
+                        for await status in tour.$status {
+                            tourStatus = status
+                        }
+                    }
+                } catch {
+                    self.error = error
+                }
+            }
```

Comment:
```suggestion
            }
            .onDisappear {
                tourController.pause()
            }
```
Currently the KML tour keeps playing when exiting the sample. The tour should be paused.

Reply:
It doesn't appear to continue playing for me. Does the `tourController` not fall out of scope when exiting the sample, and if not, should it just set to `nil` instead of pausing it?

Reply:
I can reproduce the issue on iOS 15 with an iPhone. Since `tourController` is non-optional I'm not sure it can be set to `nil`.